source/common/uma_pools/disjoint_pool.cpp

```diff
@@ -373,7 +375,13 @@ Slab::~Slab() {
     } catch (std::exception &e) {
         std::cout << "DisjointPool: unexpected error: " << e.what() << "\n";
     }
-    memoryProviderFree(bucket.getMemHandle(), MemPtr);
+
+    try {
+        memoryProviderFree(bucket.getMemHandle(), MemPtr);
+    } catch (MemoryProviderError &e) {
+        std::cout << "DisjointPool: error from memory provider: " << e.code
```

Comment:
is it the right way just to print to std::cout?

Reply:
We need to integrate UR logger into the disjoint pool (or implement a separate logger for UMF if we'll want to move it to separate repo) first. This is on the TODO list - there are multiple 'cout' calls int the DisjointPool implementation currently.